cmd/kube-controller-manager/app/options/options.go

```diff
@@ -434,6 +435,7 @@ func (s KubeControllerManagerOptions) Config(allControllers []string, disabledBy
 	kubeconfig.ContentConfig.ContentType = s.Generic.ClientConnection.ContentType
 	kubeconfig.QPS = s.Generic.ClientConnection.QPS
 	kubeconfig.Burst = int(s.Generic.ClientConnection.Burst)
+	kubeconfig.Timeout = 70 * time.Second // slightly bigger than the default server timeout which is 60 seconds
```

Comment:
FYI, this would put `timeout=70s` to the request URI and the apiserver would truncate it to `60s` :)

https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/rest/request.go#L494-L497

Reply:
yeah, so the requests should be timed out after 60s if not then we know there was something that was holding up requests, for example, a proxy.